tests/bpf2c_tests/expected/atomic_instruction_raw.c

```diff
@@ -0,0 +1,159 @@
+         0,                  // The id of the inner map template.
+     },
+     "map"},
+};
+#pragma data_seg(pop)
+
+static void
+_get_maps(_Outptr_result_buffer_maybenull_(*count) map_entry_t** maps, _Out_ size_t* count)
+{
+    *maps = _maps;
+    *count = 1;
+}
+
+static helper_function_entry_t func_helpers[] = {
+    {NULL, 1, "helper_id_1"},
+};
+
+static GUID func_program_type_guid = {0xf1832a85, 0x85d5, 0x45b0, {0x98, 0xa0, 0x70, 0x69, 0xd6, 0x30, 0x13, 0xb0}};
+static GUID func_attach_type_guid = {0x85e0d8ef, 0x579e, 0x4931, {0xb0, 0x72, 0x8e, 0xe2, 0x26, 0xbb, 0x2e, 0x9d}};
+static uint16_t func_maps[] = {
+    0,
+};
+
+#pragma code_seg(push, "xdp_prog")
+static uint64_t
+func(void* context)
+#line 25 "sample/atomic_instruction.c"
+{
+#line 25 "sample/atomic_instruction.c"
+    // Prologue
+#line 25 "sample/atomic_instruction.c"
+    uint64_t stack[(UBPF_STACK_SIZE + 7) / 8];
+#line 25 "sample/atomic_instruction.c"
+    register uint64_t r0 = 0;
+#line 25 "sample/atomic_instruction.c"
+    register uint64_t r1 = 0;
+#line 25 "sample/atomic_instruction.c"
+    register uint64_t r2 = 0;
+#line 25 "sample/atomic_instruction.c"
+    register uint64_t r3 = 0;
+#line 25 "sample/atomic_instruction.c"
+    register uint64_t r4 = 0;
+#line 25 "sample/atomic_instruction.c"
+    register uint64_t r5 = 0;
+#line 25 "sample/atomic_instruction.c"
+    register uint64_t r10 = 0;
+
+#line 25 "sample/atomic_instruction.c"
+    r1 = (uintptr_t)context;
+#line 25 "sample/atomic_instruction.c"
+    r10 = (uintptr_t)((uint8_t*)stack + sizeof(stack));
+
+    // EBPF_OP_MOV64_IMM pc=0 dst=r1 src=r0 offset=0 imm=0
+#line 25 "sample/atomic_instruction.c"
+    r1 = IMMEDIATE(0);
+    // EBPF_OP_STXDW pc=1 dst=r10 src=r1 offset=-8 imm=0
+#line 27 "sample/atomic_instruction.c"
+    *(uint64_t*)(uintptr_t)(r10 + OFFSET(-8)) = (uint64_t)r1;
+    // EBPF_OP_MOV64_REG pc=2 dst=r2 src=r10 offset=0 imm=0
+#line 27 "sample/atomic_instruction.c"
+    r2 = r10;
+    // EBPF_OP_ADD64_IMM pc=3 dst=r2 src=r0 offset=0 imm=-8
+#line 27 "sample/atomic_instruction.c"
+    r2 += IMMEDIATE(-8);
+    // EBPF_OP_LDDW pc=4 dst=r1 src=r0 offset=0 imm=0
+#line 28 "sample/atomic_instruction.c"
+    r1 = POINTER(_maps[0].address);
+    // EBPF_OP_CALL pc=6 dst=r0 src=r0 offset=0 imm=1
+#line 28 "sample/atomic_instruction.c"
+    r0 = func_helpers[0].address
+#line 28 "sample/atomic_instruction.c"
+         (r1, r2, r3, r4, r5);
+#line 28 "sample/atomic_instruction.c"
+    if ((func_helpers[0].tail_call) && (r0 == 0))
+#line 28 "sample/atomic_instruction.c"
+        return 0;
+    // EBPF_OP_JEQ_IMM pc=7 dst=r0 src=r0 offset=2 imm=0
+#line 29 "sample/atomic_instruction.c"
+    if (r0 == IMMEDIATE(0))
+#line 29 "sample/atomic_instruction.c"
+        goto label_1;
+    // EBPF_OP_MOV64_IMM pc=8 dst=r1 src=r0 offset=0 imm=1
+#line 29 "sample/atomic_instruction.c"
+    r1 = IMMEDIATE(1);
+    //  pc=9 dst=r0 src=r1 offset=0 imm=0
```

Comment:
Can you update it to show the instruction being emulated in the debug info here?

Reply:
fixed.